api/python/cellxgene_census/src/cellxgene_census/experimental/diffexp/memento/diff_expr.py

```diff
@@ -79,12 +120,33 @@ def compute_all(
     )
 
     # make treatment variable be in the first column of the design matrix
-    variables = [treatment] + [covariate for covariate in CUBE_LOGICAL_DIMS_OBS if covariate != treatment]
-    design = pd.get_dummies(obs_groups_df[variables].astype(str), drop_first=True, dtype=int)
-    assert design.shape[1] == obs_groups_df[variables].nunique().sum() - len(variables)
+    variables = [treatment] + [covariate for covariate in covariates if covariate != treatment]
+
+    agg_dict = {i: "first" for i in variables}
+    agg_dict["n_obs"] = "sum"
+    obs_group_joinid_agg_series = obs_groups_df.groupby("obs_group_joinid").apply(
```

Comment:
I'm not sure I follow why you need to group by `obs_group_joinid`, since that is already unique.